src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -722,9 +725,31 @@ private List<LaunchCommand> GetInitializeCommands()
                     {
                         commands.Add(new LaunchCommand("-exec-arguments " + _launchOptions.ExeArguments));
                     }
+                    
+                    Func<string, Task> breakMainSuccessHandler = (string bkptResult) =>
+                    {
+                        int index = bkptResult.IndexOf("number=", StringComparison.Ordinal);
+                        if (index > 0)
+                        {
+                            string trimmedInnerText = bkptResult.Substring(index).Trim('\r', '\n', '{', '}');
+                            Dictionary<string, string> dict = trimmedInnerText
+                                .Split(',')
+                                .Select(value => value.Split('='))
+                                .Where(x => x.Length == 2)
+                                .ToDictionary(x => x.First(), x => x.Last());
+
+                            if (dict.Keys.Contains("number"))
```

Comment:
in case of bad split? I'm worried about trying to access a key that doesn't exist and it throwing an exception. This seemed safer.